src/databricks/labs/ucx/progress/install.py

```diff
@@ -0,0 +1,75 @@
+import datetime as dt
+import logging
+from dataclasses import dataclass
+
+from databricks.labs.lsql.backends import Dataclass, SqlBackend
+from databricks.sdk.errors import InternalError
+from databricks.sdk.retries import retried
+
+from databricks.labs.ucx.framework.utils import escape_sql_identifier
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class HistoricalRecord:
+    workspace_id: int
+    """The identifier of the workspace where this record was generated."""
+
+    run_id: int
+    """An identifier of the workflow run that generated this record."""
+
+    snapshot_id: int
+    """An identifier that is unique to the records produced for a given snapshot."""
+
+    run_start_time: dt.datetime
+    """When this record was generated."""
+
+    object_type: str
+    """The inventory table for which this record was generated."""
+
+    object_type_version: int
+    """Versioning of inventory table, for forward compatibility."""
+
+    object_id: list[str]
+    """The type-specific identifier for this inventory record."""
+
+    object_data: str
+    """Type-specific JSON-encoded data of the inventory record."""
+
+    object_owner: str
```

Comment:
```suggestion
    owner: str
```
is it user ID or user name? 
https://databricks-sdk-py.readthedocs.io/en/latest/dbdataclasses/iam.html#databricks.sdk.service.iam.User.id

Reply:
also add application state migration

Reply:
> is it user ID or user name?

What is more consistent incl. programmatic users? I prefer user name, but if that does not work well for SP and 
IAM, then id

Reply:
> also add application state migration

What do you refer to with this?

Reply:
I've also just surveyed all our inventory classes. Of the 19 classes, only 5 currently have a creator field:
 * `ClusterInfo`: `creator`
 * `GlobalInitScriptInfo`: `script_id`
 * `JobInfo`: `creator`
 * `PipelineInfo`: `creator_name`
 * `PolicyInfo`: `creator`

All of these are optional.

Reply:
Following some discussion, we've concluded that all classes that end up in the history table need to have an associated owner. This will require class-specific logic. When there is no logical (or available) owner, the workspace admin should be designated as owner.

Reply:
Note: #2761 has been created to track the owner logic.

Reply:
Owner logic has landed in #2774.

Reply:
owner is tracked separately

Reply:
i meant "populating the owner" was tracked separately